src/game.cpp

```diff
@@ -12097,10 +12097,22 @@ bool game::plmove(int dx, int dy, int dz)
     }
 
     // Invalid move
-    const bool waste_moves = u.has_effect( effect_blind ) || u.worn_with_flag("BLIND") || u.has_active_bionic("bio_blindfold") || u.has_effect( effect_stunned );
+    const bool waste_moves = u.is_blind() || u.has_active_bionic("bio_blindfold") || u.has_effect( effect_stunned );
     if( waste_moves || dest_loc.z != u.posz() ) {
+        std::string obtacle_name;
+        int part;
+        vehicle *veh = m.veh_at( dest_loc, part );
+        if( veh != nullptr ) {
+            // redefine variable as id of obtacle part
+            part = veh->obstacle_at_part( part );
```

Comment:
That functions name is kind of inconsistent, it should be `obtacle_at_part`. Or change "obtacle" in the variable name to "obstacle". Just for consistency.